### PR TITLE
🐛 Source S3: Fix discovery for zip file

### PR DESCRIPTION
**Pull request by tolik0**
## What

Fix the discovery for zip file. Resolves: https://github.com/airbytehq/oncall/issues/3526

## How

Updated the file check to correctly identify files ending with `.zip` extension, rather than just `zip`.

### CODE REVIEW

File: airbyte-integrations/connectors/source-s3/source_s3/v4/stream_reader.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def _page(
                 break
 
     def _handle_file(self, file):
-        if file["Key"].endswith("zip"):
+        if file["Key"].endswith(".zip"):
```
Comment (girarda) [inline airbyte-integrations/connectors/source-s3/source_s3/v4/stream_reader.py:160]: what problem does this solve? did we encounter files ending in `zip` that did not have an extension?

Comment (tolik0) [reply airbyte-integrations/connectors/source-s3/source_s3/v4/stream_reader.py:160]: @girarda Yes, the sync failed due to the file without an extension, ending with 'zip'.

File: docs/integrations/sources/s3.md [hunk 2]
```diff
@@ -256,7 +256,8 @@ To perform the text extraction from PDF and Docx files, the connector uses the [
 
 | Version | Date       | Pull Request                                                                                                    | Subject                                                                                                              |
 |:--------|:-----------|:----------------------------------------------------------------------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------|
-| 4.2.1 | 2023-11-13 | [32357](https://github.com/airbytehq/airbyte/pull/32357) | Improve spec schema |
+| 4.2.2   | 2023-11-20 | [32677](https://github.com/airbytehq/airbyte/pull/32677)                                                        | Fix discovery for zip file                                                                                           |
```
Comment (girarda) [inline docs/integrations/sources/s3.md:259]:
```suggestion
| 4.2.2   | 2023-11-20 | [32677](https://github.com/airbytehq/airbyte/pull/32677)                                                        | Only read files with ".zip" extension as zipped files                                                                                       |
```

**Review by girarda — APPROVED** [review]

File: docs/integrations/sources/s3.md (hunk 2 above)
Comment (girarda) [reply docs/integrations/sources/s3.md:259]: suggestion for a clearer error message since the change is not related to discovery

**Comment by octavia-approvington** [comment]
This code is at peace with itself <br/> imagine karate kid approving

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**